File: 0143-reorder-list/0143-reorder-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
void reorderList(struct ListNode* head) {
   struct ListNode *slow=head,*fast=head;
        while(fast!=NULL&&fast->next!=NULL){
            slow=slow->next;
            fast=fast->next->next;
        }
        struct ListNode *cur=slow->next,*prev=NULL;
        slow->next=NULL;
        while(cur){
            struct ListNode *next=cur->next;
            cur->next=prev;
            prev=cur;
            cur=next;

        }
        struct ListNode *head2=prev,*head1=head;
        while(head2){
           struct ListNode *temp1=head1->next;
           struct ListNode *temp2=head2->next;
           head1->next=head2;
           head2->next=temp1;
           head1=temp1;
           head2=temp2;
        }

        
}

File: 0143-reorder-list/0143-reorder-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public void reorderList(ListNode head) {
        ListNode fast=head,slow=head;
        while(fast!=null&&fast.next!=null){
            slow=slow.next;
            fast=fast.next.next;
        }
        ListNode temp=slow,next=null,prev=null;
        while(temp!=null){
            next=temp.next;
             temp.next=prev;
             prev=temp;
             temp=next;
        }
        ListNode first=head,second=prev;
        while(second.next!=null){
            ListNode temp1=first.next;
            ListNode temp2=second.next;

            first.next=second;
            second.next=temp1;

            first=temp1;
            second=temp2;
        }
        
    }
}